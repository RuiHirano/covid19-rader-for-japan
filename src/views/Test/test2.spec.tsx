import * as React from "react";
import {
    cleanup,
    fireEvent,
    getByLabelText,
    getByTestId
} from "@testing-library/react";
import { configure, shallow, mount } from "enzyme";
import { render } from "@testing-library/react";
import Test from "./Test";
import { Button, TextField, Typography } from "@material-ui/core";
import { createShallow, createMount } from "@material-ui/core/test-utils";
import { act } from "react-dom/test-utils";

afterEach(cleanup);

describe("スクリーンショットテスト", () => {
    it("Snapshot of <Test />", () => {
        const { asFragment } = render(<Test />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("機能テスト", () => {
    let shallow: any;
    let mount: any;
    beforeEach(() => {
        shallow = createShallow();
        mount = createMount();
    });

    it("ButtonをおすとCountが増える", () => {
        const wrapper = shallow(<Test />);
        const button = wrapper.find(Button);
        const text = wrapper.find(Typography);
        console.log("wrapper ", wrapper.debug());
        console.log("button ", button.debug());
        console.log("text ", text.debug());
        expect(text.text()).toBe("0");
        button
            .at(0)
            .props()
            .onClick();
        expect(text.text()).toBe("1");
    });

    it("TextField", () => {
        let wrapper: any;
        act(() => {
            wrapper = shallow(<Test />);
        });
        let textField = wrapper.find(TextField);
        console.log("debug", textField.debug());

        textField.simulate("change", { target: { value: "test" } });
        /*textField
            .at(0)
            .props()
            .onChange({ target: { value: "test" } });*/
        wrapper.update();
        console.log("debug2", textField.debug());
        expect(textField.text()).toBe("test");
    });
});
