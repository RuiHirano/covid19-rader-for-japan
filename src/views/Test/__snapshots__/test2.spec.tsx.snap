// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スクリーンショットテスト Snapshot of <Test /> 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      >
        Standard
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          data-testid="standard"
          type="textarea"
          value=""
        />
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      0
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Button
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
